src/main.rs

```diff
@@ -176,70 +245,32 @@ fn check_rust_version() {
     }
 }
 
-fn build_elf(command: &str, args: &[&str]) {
-    let rustflags = env::var("RUSTFLAGS").unwrap_or_default()
-        + &format!(" -L{}/libctru/lib -lctru", env::var("DEVKITPRO").unwrap());
-
-    let mut process = Command::new("cargo")
-        .arg(command)
-        .arg("-Z")
-        .arg("build-std")
-        .arg("--target")
-        .arg("armv6k-nintendo-3ds")
-        .args(args)
-        .env("RUSTFLAGS", rustflags)
-        .stdin(Stdio::inherit())
-        .stdout(Stdio::inherit())
-        .stderr(Stdio::inherit())
-        .spawn()
-        .unwrap();
-
-    let status = process.wait().unwrap();
-
-    if !status.success() {
-        process::exit(status.code().unwrap_or(1));
-    }
-}
-
-fn get_metadata(args: &[&str], opt_level: &str) -> CTRConfig {
+fn get_metadata(messages: &[Message]) -> CTRConfig {
     let metadata = MetadataCommand::new()
         .exec()
         .expect("Failed to get cargo metadata");
-    let target_dir = &metadata.target_directory;
-
-    let package: &Package;
-    let binary_name: String;
-    let target_path: String;
-
-    // Check if we compiled the crate or an example
-    if let Some(example_pos) = args.iter().position(|arg| *arg == "--example") {
-        let example_name = *args.get(example_pos + 1).expect("No example given");
-
-        // Find the example's package
-        package = metadata
-            .packages
-            .iter()
-            .find(|pkg| {
-                pkg.targets.iter().any(|target| {
-                    target.name == example_name && target.kind.iter().any(|kind| kind == "example")
-                })
-            })
-            .expect("Could not find package for example");
-
-        binary_name = format!("{} - {} example", example_name, package.name);
```

Comment:
What do example binary names look like now? We should keep something like this for examples at least, and do something similar for tests.

Reply:
Updated to restore the previous naming convention, and made a better name for tests.